source/slang/slang-ir-autodiff.cpp

```diff
@@ -133,12 +133,13 @@ bool isNoDiffType(IRType* paramType)
     }
 
     while (auto ptrType = as<IRPtrTypeBase>(paramType))
-        paramType = ptrType->getValueType();
-    while (auto attrType = as<IRAttributedType>(paramType))
     {
-        if (attrType->findAttr<IRNoDiffAttr>())
+        paramType = ptrType->getValueType();
```

Comment:
I think it should be a general loop that can peel off either an attributed type or a ptr type during each iteration.